app/uk/gov/hmrc/lisaapi/services/TransactionService.scala

```diff
@@ -147,22 +156,24 @@ class TransactionService @Inject()(desConnector: DesConnector)(implicit ec: Exec
           bonusDueForPeriod = bonusDueForPeriod
         )
       }
-      case error: DesFailureResponse => getFailureResponse(error.code, transactionId, bonusDueForPeriod)
-    }
-  }
-
-  private def getFailureResponse(errorCode: String, transactionId: String, bonusDueForPeriod: Option[Amount] = None): GetTransactionResponse = {
-    errorCode match {
-      case "NOT_FOUND" =>
+      case error: DesFailureResponse if error.code == "COULD_NOT_PROCESS" => {
```

Comment:
Is it worth combining these into a case match on 'error.code'?

Reply:
Good shout. Pushing a new commit to do just that.